AsyncExpiringLazy/EagerAsyncExpiringLazy.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using AsyncExpiringLazy;
+
+namespace Strathweb
+{
+    /// <summary>
+    /// Eager implementation of the <see cref="AsyncExpiringLazy{T}"/> which uses a background worker to refresh the item.
+    /// It performs better in the situations when factory method is slow and having a fast access to the fresh and ready T item is important.
+    /// The background worker is started after the first call to the <see cref="Value"/> method.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public class EagerAsyncExpiringLazy<T> : IDisposable
```

Comment:
I would prefer for this to be called `AsyncExpiringEager<T>` - since lazy and eager are mutually exclusive

Reply:
This name also came to my mind, however, the default behavior of the Lazy still exists - value is not created until it's called. Only the refresh mechanism is changed to have an extra option to replace the underlying value with the background monitor. However, from the usage perspective, AsyncExpiringEager may be a better option.